docs/articles/user_acquisition_analytics.md

```diff
@@ -0,0 +1,442 @@
+)
+umap_df = umap_df.join(label_df)
+```
+
+Next,we join our dataframes and create a chart, letting us visualize the UMAP-transformed vectors, and coloring them with cluster labels.
+
+```python
+umap_df = umap_df.join(label_df)
+alt.Chart(umap_df).mark_circle(size=8).encode(
+    x="dimension_1", y="dimension_2", color="cluster_label:N"
+).properties(
+    width=600, height=500, title="UMAP Transformed vectors coloured by cluster labels"
+).configure_title(
+    fontSize=16,
+    anchor="middle",
+).configure_legend(
+    strokeColor="black",
+    fillColor="#EEEEEE",
+    padding=10,
+    cornerRadius=10,
+    labelFontSize=14,
+    titleFontSize=14,
+).configure_axis(
+    titleFontSize=14, labelFontSize=12
+)
+```
+
+![cluster visualization](../assets/use_cases/user_acquisition_analytics/cluster_visualization.png)
+
+The dark blue clusters (label -1) are outliers - not large or dense enough to form a distinct group. The large number of blobs indicates that 2/3 of the vector normal mass is comprised of *many* ad_creatives, while 1/3 is made up by just a *few*.
+*Note - 2D (UMAP) visualizations often make some clusters look quite dispersed / scattered.
+
+## Understanding the cluster groups
+
+To understand our user clusters better, we can generate some activity histograms.
+
+First, we join user data with cluster labels, create separate DataFrames for each cluster, generate activity histograms for each cluster, and then concatenate these histograms into a single visualization.
+
+```python
+# activity histograms by cluster
+user_df = user_df.set_index("id").join(label_df)
+
+by_cluster_data = {
+    label: user_df[user_df["cluster_label"] == label]
+    for label in np.unique(hdbscan.labels_)
+}
+
+activity_histograms = [
+    alt.Chart(user_df_part)
+    .mark_bar()
+    .encode(x=alt.X("activity", bin=True), y="count()")
+    .properties(title=f"Activity histogram for cluster {label}")
+    for label, user_df_part in by_cluster_data.items()
+]
+
+alt.hconcat(*activity_histograms)
+```
+
+![histograms of clusters](../assets/use_cases/user_acquisition_analytics/cluster_histograms.png)
+
+From our histograms, we can observe that:
+
+- cluster -1, though predominantly low activity users, has an unusually large number of extremely active users (outliers)
+- cluster 0 is largely medium activity users, with no high activity users
+- cluster 1 users are moderately active, with relatively few very low and high activity users
+- clusters 2 and 3 are predominately low activity with no high activity users
+
+To see the distribution of ad_creatives across different clusters, we create a DataFrame that shows each ad_creative's count value within each cluster:
+
+```python
+pd.DataFrame(user_df.groupby("cluster_label")["ad_creative"].value_counts())
+```
+
+![ad creatives distribution across clusters](../assets/use_cases/user_acquisition_analytics/ad_creatives-per-cluster.png)
+
+observations:
+
```

Comment:
@morkapronczay
This needs updating in the notebook and google colab to reflect correction of previous error.
So does the summary table at the bottom. (For this article, I'm omitting the summary table below, replacing with a new markdown table - maybe can use that once it's done.)

Reply:
I'll update the notebook removing the note